__tests__/app.test.js

```diff
@@ -324,6 +324,44 @@ describe('POST /api/articles/:article_id/comments', () => {
     })
 
 });
+describe('add query functionality GET /api/articles ', () => {
+    test('Should respond with a an appropriately sorted array of articles', () => {
+        return request(app)
+        .get('/api/articles?sort_by=created_at&order=ASC')
+        .expect(200)
+        .then(({body: {articles}})=> {
+            expect(articles).toBeSortedBy("created_at", {
+                ascending: true,
+            });
+        });
+    });
+    test('Should respond with a array of articles filtered by topic if given a topic query', () => {
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({body: {articles}}) => {
+            expect(articles).toHaveLength(1)
+            expect(articles[0].topic).toEqual('cats')
+        })
+    })
+    test('Should respond with a status 400 if a query has a invalid column name to sort by', () => {
+            return request(app)
+            .get('/api/articles?sort_by=nonsense')
+            .expect(400)
+            .then(({body: {msg}})=> {
+                expect(msg).toBe('Bad Request')
+            });
+    });
+    test('Should respond with a status 400 if a query has a invalid string to order by but a valid column name to sort by', () => {
+        return request(app)
+        .get('/api/articles?sort_by=author&order=notDesc')
+        .expect(400)
+        .then(({body: {msg}})=> {
+            expect(msg).toBe('Bad Request')
+        });
+
+});      
+})
```

Comment:
There are few additional tests it would be good to see for the topic! What if the topic given is invalid or not found? Or if there are no articles for a topic that exists?